elyra/pipeline/processor.py

```diff
@@ -116,11 +126,9 @@ def object_storage_path(self):
         """
         return self._object_storage_path
 
+    @abstractmethod
     def to_json(self):
-        return {"run_url": self.run_url,
-                "object_storage_url": self.object_storage_url,
-                "object_storage_path": self.object_storage_path
-                }
+        raise NotImplementedError()
```

Comment:
I would leave this method as-is, not make it abstract, but add `"platform": self.type,` to the result.  The KFP subclass would then not need to implement `to_json()`, while the Airflow subclass would call super().to_json(), then add in its `git_url` value.